### PR TITLE
pkg/trace/{api,pb}: update model and add obfuscation

### PR DESCRIPTION
**Pull request by gbbr**
This change updates the model to the latest discussed one and enables
obfuscation.

### CODE REVIEW

File: pkg/trace/api/api.go [hunk 1]
```diff
@@ -410,51 +422,35 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
...
+				if err != nil {
+					log.Errorf("Error obfuscating resource %q: %v", b.Resource, err)
+				} else {
+					b.Resource = oq.Query
 				}
 			}
 
+			tags := map[string]string{"version": in.Version}
+			if b.HTTPStatusCode != 0 {
+				tags["http.status_code"] = strconv.Itoa(int(b.HTTPStatusCode))
```
Comment (gbbr) [inline pkg/trace/api/api.go:443]: Note that [this](https://github.com/DataDog/datadog-agent/blob/18565f2c5cf9c28d4f52ed0b6930384b082f3972/pkg/trace/agent/normalizer.go#L162) was the previous validation happening on the HTTP status code. I think there's no need for it now that we're an `int`.

File: pkg/trace/api/api.go [hunk 2]
```diff
@@ -467,37 +463,34 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
...
-				Key:      key,
-				Name:     b.Name,
-				Measure:  stats.HITS,
-				TagSet:   tagset,
-				TopLevel: b.TopLevel,
-				Value:    b.Hits,
+				Key:     key,
+				Name:    b.Name,
+				Measure: stats.HITS,
+				TagSet:  tagset,
```
Comment (furmmon) [inline pkg/trace/api/api.go:476]: you still need to set TopLevel on the old payload,
you can set it at Hits value
TopLevel: stats.HITS

File: pkg/trace/api/api.go [hunk 3]
```diff
@@ -401,6 +402,17 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
...
 	}
 	metrics.Count("datadog.trace_agent.receiver.stats_buckets", int64(len(in.Stats)), nil, 1)
+
+	if in.Env == "" {
+		in.Env = r.conf.DefaultEnv
+	}
+	in.Env = traceutil.NormalizeTag(in.Env)
+	for _, rule := range r.conf.ReplaceTags {
+		switch rule.Name {
+		case "env", "*":
```
Comment (furmmon) [inline pkg/trace/api/api.go:412]: The version case is removed in this PR, I don't know if it's actually used

File: pkg/trace/api/api.go [hunk 4]
```diff
@@ -410,51 +422,35 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
...
-				}
-				if len(parts) > 1 {
-					if v := parts[1]; len(v) > traceutil.MaxMetaValLen {
-						parts[1] = traceutil.TruncateUTF8(v, traceutil.MaxMetaValLen) + "..."
-					}
-					tags[parts[0]] = parts[1]
+			if b.Type == "sql" || b.DBType != "" {
+				// TODO(gbbr): perhaps we should store only one instance instead of
+				// a new one on each request?
+				oq, err := obfuscate.NewObfuscator(nil).ObfuscateSQLString(b.Resource)
```
Comment (furmmon) [inline pkg/trace/api/api.go:433]: Is this the only obfuscator applying to resources?

File: pkg/trace/api/api.go (hunk 3 above)
Comment (gbbr) [reply pkg/trace/api/api.go:412]: It will automatically be applied to the `tags` var defined in the `for` loop

File: pkg/trace/api/api.go (hunk 4 above)
Comment (gbbr) [reply pkg/trace/api/api.go:433]: yes

File: pkg/trace/api/api.go (hunk 2 above)
Comment (gbbr) [reply pkg/trace/api/api.go]: Done. PTAL

**Review by raphaelgavache — APPROVED** [review]

**Pull request merged**